### PR TITLE
Added multi filter support for BuildingFilter !!

### PR DESCRIPTION
**Pull request by woo1127**
<b>Finally</b> Here it comes !!!  BuildingFilter can now include techFilter to form multiple filters within a single unique, similar to baseUnitFilter. While it might seem like a simple fix, it actually took me several days to thoroughly read and understand the entire unique system.😖🥴

So Now, the problem regarding the 'Monument to the Gods' pantheon and 'Marbles' has finally been resolved. Please refer to the attached references below :
- https://civilization.fandom.com/wiki/Marble_(Civ5)
- https://civilization.fandom.com/wiki/Religion_(Civ5)#Pantheons

The updated buildingFilter has been tested with the following uniques :

`"[+11]% Production when constructing [{Faith} {Calendar} {Ancient era}] wonders [in all cities]", // stonehenge`
`"[+12]% Production when constructing [{Science} {Writing} {Ancient era}] wonders [in all cities]", // great library`
`"[+13]% Production when constructing [{Science} {Writing}] buildings [in all cities]", // library`
`"[+14]% Production when constructing [{Ancient era} {Food}] buildings [in all cities]", // granary and water mill`
`"[+15]% Production when constructing [{non-[Culture]} {non-[Food]} {Faith} {Ancient era}] buildings [in all cities]", // shrine`

### CODE REVIEW

**Comment by yairm210** [comment]
What about the system was hard to understand?
I want to write it up for clarity

**Review by yairm210 — APPROVED** [review]
All in all looks good!

File: android/assets/jsons/Civ V - Gods & Kings/Policies.json [hunk 1]
```diff
@@ -84,7 +84,8 @@
                 "name": "Republic",
                 "uniques": [
                     "[+1 Production] [in all cities]",
-                    "[+5]% Production when constructing [All] buildings [in all cities]"
+                    "[+5]% Production when constructing [All] buildings [in all cities]",
+                    "[+5]% Production when constructing [All] wonders [in all cities]"
```
Comment (SeventhM) [inline android/assets/jsons/Civ V - Gods & Kings/Policies.json:88]: Needs confirmation from Civ V. There's no real documentation that this is supposed to actually give production for wonders

File: android/assets/jsons/Civ V - Vanilla/Policies.json [hunk 2]
```diff
@@ -103,7 +103,8 @@
                 "name": "Republic",
                 "uniques": [
                     "[+1 Production] [in all cities]",
-                    "[+5]% Production when constructing [All] buildings [in all cities]"
+                    "[+5]% Production when constructing [All] buildings [in all cities]",
+                    "[+5]% Production when constructing [All] wonders [in all cities]"
```
Comment (SeventhM) [inline android/assets/jsons/Civ V - Vanilla/Policies.json:107]: Same here

File: android/assets/jsons/Civ V - Vanilla/Policies.json [hunk 3]
```diff
@@ -219,6 +220,7 @@
         },
         "uniques": [
             "[+15]% Production when constructing [Culture] buildings [in all cities]",
+            "[+15]% Production when constructing [Culture] wonders [in all cities]",
```
Comment (SeventhM) [inline android/assets/jsons/Civ V - Vanilla/Policies.json:223]: Same here

File: android/assets/jsons/Civ V - Vanilla/Policies.json [hunk 4]
```diff
@@ -655,6 +657,7 @@
                 "uniques": [
                     "[+1 Production] [in all cities]",
                     "[+10]% Production when constructing [All] buildings [in all cities]",
+                    "[+10]% Production when constructing [All] wonders [in all cities]",
```
Comment (SeventhM) [inline android/assets/jsons/Civ V - Vanilla/Policies.json:660]: Same here

**Comment by SeventhM** [comment]
Yoooooooo. We're finally allowing eras for buildings to remove the one comment. Nice

My main hold up are edits to wonder production that I'm not sure is actually the case in Civ 5. Otherwise, I'll approve

File: android/assets/jsons/Civ V - Gods & Kings/Policies.json (hunk 1 above)
Comment (SeventhM) [reply android/assets/jsons/Civ V - Gods & Kings/Policies.json:88]: And there's basically no guides or whatever that suggests this does give production to wonders, unlike other niche things

**Comment by woo1127** [comment]
> What about the system was hard to understand? I want to write it up for clarity

Actually I would say ALL !! However, that is not your concern, you and other contributors have done an excellent job 🤩!! The issue might stem from my lack of knowledge about Kotlin, Libgdx, and my first encounter with a large codebase.

Now there are few questions in my mind now I would like to understand, what does clone() and setTransient() function actually do and how can I know in what scenario I would need to called them ?

**Comment by woo1127** [comment]
I have removed all the lines regarding to wonder production first to ensure that this PR can be safely merged in next version. Further update will continue on #11304

**Pull request merged**